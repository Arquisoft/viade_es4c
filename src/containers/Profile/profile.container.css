.profile-img {
    height: 20px;
}

.accordion {
    margin-top: 25px;
}

.collapse > ul {
    list-style: none;
    font-size: 1rem;
    margin-top: 5px;
    margin-left: -30px;
}

.card-text {
    margin-top: 25px;
}

.input-group > .input-group-append > .btn {
    background: #34d1ab;
    border-color: #34d1ab;
    color: #f2f2f2;
}