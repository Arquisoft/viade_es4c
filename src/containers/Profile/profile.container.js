import React from "react";
import {List, Value, withWebId} from "@solid/react";
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";
import {Image} from "react-bootstrap";
import {Spacer} from "../../components";
import Container from "react-bootstrap/Container";
import Card from "react-bootstrap/Card";
import "./profile.container.css";
import Accordion from "react-bootstrap/Accordion";
import Button from "react-bootstrap/Button";
import {Link} from "react-router-dom";

class ProfileComponent extends React.Component {
    state = { profileInput: "", activeProfile: "" };

    componentDidUpdate(prevProps) {
        const webId = this.props.webId;
        if (webId && webId !== prevProps.webId) {
            this.setState({profileInput: webId});
            this.setState({activeProfile: webId});
        }
    }

    viewProfile(profile) {
        this.setState({ profileInput: profile, activeProfile: profile });
    }

    render() {
        const profileInput = this.state.profileInput;
        const activeProfile  = this.state.activeProfile;
        return (
            <Container>
                {/* Spacer row */}
                <Row>
                    <Col xs={12} md={12}>
                        <Spacer/>
                    </Col>
                </Row>
                {/* Description row */}
                <Row>
                    <Col xs={12}/>
                    <Col xs={12} md={8} className="profile-card">
                        <Card className="text-center">
                            <Card.Header>
                                <Image src="/viade_es4c/img/profile/user.png" className="profile-img"/>
                                <Value src={`[${activeProfile}].name`}/>
                            </Card.Header>
                            <Card.Body>
                                <Card.Title>
                                    <Image src="/viade_es4c/img/Solid.svg" className="profile-img"/>
                                    <a href={profileInput}>
                                        {profileInput}
                                    </a>
                                </Card.Title>
                                <Accordion defaultActiveKey="0">
                                    <Card>
                                        <Card.Header>
                                            <Accordion.Toggle as={Button} variant="link" eventKey="0">
                                                Friends
                                            </Accordion.Toggle>
                                        </Card.Header>
                                        <Accordion.Collapse eventKey="0">
                                            <List src={`[${activeProfile}].friends`}>{(friend) =>
                                                <li key={friend}>
                                                    <Value src={`[${friend}].name`}>{`${friend}`}</Value>
                                                </li>}
                                            </List>
                                        </Accordion.Collapse>
                                    </Card>
                                </Accordion>
                                <Card.Text>
                                    <Link to="/myRoutes">
                                        <Image src="/viade_es4c/img/home/keepyours.svg" className="profile-img"/>
                                        <h2>My routes</h2>
                                    </Link>
                                </Card.Text>
                            </Card.Body>
                        </Card>
                    </Col>
                    <Col xs={12}/>
                </Row>
            </Container>
        );
    }
}
export default withWebId(ProfileComponent);