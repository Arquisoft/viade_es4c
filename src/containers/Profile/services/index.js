import {addFriend} from "./friends.service";

export default addFriend;