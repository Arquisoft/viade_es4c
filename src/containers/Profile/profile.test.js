import React from 'react';
import { shallow } from 'enzyme';
import Profile from "./profile.container";

describe('Profile', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Profile debug />);
  
    expect(component).toMatchSnapshot();
  });
});
