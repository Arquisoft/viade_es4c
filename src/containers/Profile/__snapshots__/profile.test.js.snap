// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile should render correctly in "debug" mode 1`] = `
<ProfileComponent
  debug={true}
/>
`;
